### DISCUSSION TITLE
Make calendar colours in the Calendar Customizable

### DISCUSSION THREAD

**Discussion by Fuchs** [home-assistant/frontend#11262] [Other feature requests]
## The request

Currently it seems that calendar colours are decided at random. 
It would be nice if a custom calendar colour could be set, even if it has to be done manually in the customize yaml for the calendar object. After a quick glance at the code I tried to set backgroundColor on the calendar object, which seems to have no effect.

## The alternatives

Alternatively it would be nice if colours specified by the calendar (e.g. caldav allows calendars to have a certain colour) would be used.

## Additional information

Home Assistant 0.110.1, calendars fetched via CalDAV

**Comment by bverkron** [comment]
Yes please! Having them arbitrarily chosen and not matching what I have in Google Calendar or iCloud is very annoying and confusing for the family. At minimum using the same colours defined in Google would be great. I *think* that's possible as IIRC even back on older version of iOS it could automatically use the same calendar colours as were pulled from the Google Calendar when setup.

**Comment by zsarnett** [comment]
This should be done on the entity registry probably and made available for more than just the calendar entities

**Comment by bverkron** [comment]
Are there other entities that have colours associated with them?

**Comment by zsarnett** [comment]
I don't believe so, yet. We talked today and so far adding it to the entity attributes seems to be the best option.

**Comment by bverkron** [comment]
FYI for users in the meantime. Seems the calendars added to the calendar card entity are always assigned colours in the same order.

Example: 
```
1st = red
2nd = green
3rd = blue
4th = purple
etc...
```

So you can kind of hack around it by ordering the calendars in the card to match your color preferences. Assuming you have enough of them and can line them up the way you want. Some dummy calendars might be needed in between to increment you to the next desired colour if that makes sense. Something to keep in mind in the interim. YMMV.

**Comment by sym0nd0** [comment]
Not sure why this was changed to a discussion, rather than the feature request. 😕 

With this announcement that the Garbage Collection integration is being deprecitated, following the introduction of Local Calendar, a lot of people, including me, are now looking to replicate the same functionality that the Garbage Collection integration has provided for years but through Local Calendars.

The ability to choose the colours we need, for each calendar, is a basic requirement and this feature request would be a great start. However, [this feature request](https://community.home-assistant.io/t/local-calendar-enhancements/510905) on the forums would really enhance the useability of the entire Local Calendar integration.

**Reply by Misiu** [reply]
Take a look at the architecture discussion: https://github.com/home-assistant/architecture/discussions/883.
Another vote/opinion would be very useful and could help with moving things forward a decision on how to implement this.

**Comment by Brtrnd** [comment]
I voted for the issue and offered a way this could be done with very little changes to the code (presuming my pseudo-code is correct)

**Comment by SetMonkey13** [comment]
Putting my vote in for this to happen.  Not having the calendar match what my google calendar makes it hard to follow.  I have some calendars that I keep the same color so I can see things at a quick glance.

**Comment by kaiserkiwi** [comment]
Oh wow. I thought I'm blind not finding this option. But seeing, that this is a feature request from five years ago is really surprising. 

I wonder what's the challenge with this.

**Reply by Misiu** [reply]
I have a PR for core to support colors, so this is work in progress :)

**Reply by kaiserkiwi** [reply]
Awesome. So I had the perfect timing to integrate my calendars now. 😁
Looking forward to it.
